cylc/flow/task_pool.py

```diff
@@ -2133,24 +2122,10 @@ def _force_trigger(self, itask):
                 itask.tdef.next_point(itask.point),
                 itask.flow_nums
             )
-        # Task may be set running before xtrigger is satisfied,
-        # if so check/spawn if xtrigger sequential.
-        elif (
-            itask.is_xtrigger_sequential
-            and (
-                itask.identity not in
-                self.xtrigger_mgr.sequential_has_spawned_next
-            )
-        ):
-            self.xtrigger_mgr.sequential_has_spawned_next.add(
-                itask.identity
-            )
-            self.spawn_to_rh_limit(
-                itask.tdef,
-                itask.tdef.next_point(itask.point),
-                itask.flow_nums
-            )
         else:
+            # Task may be set running before xtrigger is satisfied,
+            # if so check/spawn if xtrigger sequential.
+            self.check_spawn_psx_task(itask)
             # De-queue it to run now.
             self.task_queue_mgr.force_release_task(itask)
```

Comment:
This has changed the logic - was it meant to? Or should it be
```python
else:
    # Task may be set running before xtrigger is satisfied,
    # if so check/spawn if xtrigger sequential.
    if not self.check_spawn_psx_task(itask):
        # De-queue it to run now.
        self.task_queue_mgr.force_release_task(itask)
```
(having changed `check_spawn_psx_task` to return a boolean)?

Reply:
Yeah, but I figured it was ok to de-queue and spawn (if psx) if not runahead released..

Reply:
Have changed it like you said... better to keep them separate I suppose, since they are different category of action..

Reply:
Actually, strike that .. This is a force trigger .. PSX tasks should be spawned outside of this runahead/de-queue here.. So have put this below.